pkg/runtime/engine/apptainer/config/config.go

```diff
@@ -660,6 +660,23 @@ func (e *EngineConfig) GetTargetGID() []int {
 	return e.JSON.TargetGID
 }
 
+// Appends all entries that are not already present in the slice.
+func AppendSliceNonduplicate(slice []string, newElements []string) []string {
```

Comment:
@upsj I see mainly two issues with the current code, first issue is that `newSlice` can allocate more memory than necessary in case of duplicate, the second issue is that `slice` is not deduplicated in the first loop with `newSlice = append(newSlice, element)`, this could be rewritten/renamed like this:
```
// ConcatenateSliceDeduplicate concatenates two string slices and returns a string slice without duplicated entries.
func ConcatenateSliceDeduplicate(first []string, second []string) []string {
	dedup := make(map[string]struct{})
	for _, slice := range [][]string{first, second} {
		for _, elem := range slice {
			dedup[elem] = struct{}{}
		}
	}
	slice := make([]string, 0, len(dedup))
	for elem := range dedup {
		slice = append(slice, elem)
	}
	return slice
}
```

Reply:
Looks much cleaner, thanks!